internal/internal_flags.go

```diff
@@ -0,0 +1,106 @@
+// The MIT License
+//
+// Copyright (c) 2023 Temporal Technologies Inc.  All rights reserved.
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files (the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+
+package internal
+
+import (
+	"math"
+
+	"go.temporal.io/api/workflowservice/v1"
+)
+
+// sdkFlag represents a flag used to help version the sdk internally to make breaking changes
+// in workflow logic.
+type sdkFlag uint32
+
+const (
+	Unset sdkFlag = iota
+	// LimitChangeVersionSASize will limit the search attribute size of TemporalChangeVersion to 2048 when
+	// calling GetVersion. If the limit is exceeded the search attribute is not updated.
+	LimitChangeVersionSASize
+	Unknown = math.MaxUint32
+)
+
+func sdkFlagFromUint(value uint32) sdkFlag {
+	switch value {
+	case uint32(Unknown):
+		return Unknown
+	case uint32(LimitChangeVersionSASize):
+		return LimitChangeVersionSASize
+	default:
+		return Unknown
+	}
+}
+
+func (f sdkFlag) isValid() bool {
+	return f != Unset && f != Unknown
+}
+
+// sdkFlags represents all the flags that are currently set in a workflow execution.
+type sdkFlags struct {
+	capabilities *workflowservice.GetSystemInfoResponse_Capabilities
+	// Flags that have been recieved from the server
+	currentFlags map[sdkFlag]bool
+	// Flags that have been set this WFT that have not been sent to the server.
+	// Keep track of them sepratly so we know what to send to the server.
+	newFlags map[sdkFlag]bool
+}
+
+func newSdkFlags(capabilities *workflowservice.GetSystemInfoResponse_Capabilities) *sdkFlags {
```

Comment:
```suggestion
func newSDKFlags(capabilities *workflowservice.GetSystemInfoResponse_Capabilities) *sdkFlags {
```

(and other places where applicable)

Reply:
I didn't name the struct `SDKFlag` because I did not want it to be exported outside internal, unless your suggesting just renaming this function?

Reply:
I am suggesting just this function and any Go function where an abbreviation needs to be capitalized (i.e. non-exported but not the first word).